api/v1beta2/ocicluster_types.go

```diff
@@ -120,6 +126,46 @@ type OCIAvailabilityDomain struct {
 	FaultDomains []string `json:"faultDomains,omitempty"`
 }
 
+// ClusterClientHostUrls contains information about client host url overrides.
+type ClusterClientHostUrls struct {
+
+	// ComputeClientHost allows the default compute SDK client URL to be changed.
+	//
+	// +optional
+	// +nullable
+	ComputeClientHost *string `json:"computeClientHost,omitempty"`
+
+	// ComputeManagementClientHost allows the default compute management SDK client URL to be changed.
+	//
+	// +optional
+	// +nullable
+	ComputeManagementClientHost *string `json:"computeManagementClientHost,omitempty"`
+
+	// VCNClientHost allows the default vcn SDK client URL to be changed.
+	//
+	// +optional
+	// +nullable
+	VCNClientHost *string `json:"vCNClientHost,omitempty"`
+
+	// LoadBalancerClientHost allows the default load balancer SDK client URL to be changed.
+	//
+	// +optional
+	// +nullable
+	LoadBalancerClientHost *string `json:"loadBalancerClientHost,omitempty"`
+
+	// IdentityClientHost allows the default identity SDK client URL to be changed.
+	//
+	// +optional
+	// +nullable
+	IdentityClientHost *string `json:"identityClientHost,omitempty"`
+
+	// ContainerEngineClientHost allows the default container engine SDK client URL to be changed.
+	//
+	// +optional
+	// +nullable
+	ContainerEngineClientHost *string `json:"containerEngineClientHost,omitempty"`
```

Comment:
in c3, if the container engine client cant be initialized, will it throw an error?

Reply:
can we test it using a random url?